### DISCUSSION TITLE
Does this project currently have automatic timing of video's transcript?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [SubtitleEdit/subtitleedit#6300] [Q&A]
Hello. I have transcription of the speech video. However it doesn't have time.

So can this project process the audio and the transcription and generate automatically timed subtitles at the moment?

Or else how can I do that? Something like YouTube does.

Thank you.

**Comment by niksedk** [comment]
Hi @FurkanGozukara,

Sorry, SE does not currently have this feature.
I guess it could be possible via speech recognition (SE currently supports vosk/kaldi).

Saw you have tried Whisper - does that work well? And do you know if it can it be compiled to an executable file?

**Reply by FurkanGozukara** [reply]
It can't be compiled into an exe file at the moment. It works fantastic though if you ask me for transcribing. I am already using it.

I even prepared 2 videos regarding Whisper

How to do Free Speech-to-Text Transcription Better Than Google Premium API with OpenAI Whisper Model : https://www.youtube.com/watch?v=msj3wuYf3d8 

This above video has raw transcriptions generated by Whisper

How Good is RTX 3060 for ML AI Deep Learning Tasks and Comparison With GTX 1050 Ti and i7 10700F CPU : https://www.youtube.com/watch?v=q8Q8CCDdSKo

Although sometimes its timing may not be perfect. Therefore I am kind of looking for perfect timing.

**Comment by niksedk** [comment]
Thx for the videos - good work :)

Whisper seems to work okay (I needed medium model to work with Danish which is rather slow - but it does work okay).
Impressive with all the languages!

I've tried to add support for Whisper in latest SE beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip
Should work on whole files + selected lines via context menu.

The timing seems to a bit off compared to Vosk/Kaldi. Do anyone know if it's possible to get one word at a time?

**Reply by ard-av** [reply]
![Screenshot_7](https://user-images.githubusercontent.com/114281218/193908274-b65da913-2a55-400e-a412-a9c955785fec.png)
![Screenshot_8](https://user-images.githubusercontent.com/114281218/193908410-ea05a205-6c65-4332-bdd2-5aed04b31e88.png)
I can’t figure out where to get and where to put the language model

**Reply by niksedk** [reply]
I guess it's not for everybody - but try this:
1) Download Python from: https://www.python.org/downloads/
2) Install pip: See https://phoenixnap.com/kb/install-pip-windows
3) Install whisper via cmd line, run: pip install git+https://github.com/openai/whisper.git 

Models can be downloaded via the SE UI or will be auto downloaded when you run whisper.... I think.

**Reply by ard-av** [reply]
So far I haven't been able to do it this way
![Screenshot_1](https://user-images.githubusercontent.com/114281218/193944574-3a2fc826-c750-4258-bad9-ece190da2453.png)
![Screenshot_2](https://user-images.githubusercontent.com/114281218/193944581-73785f81-30c5-4156-9ffc-0d63ce9e0667.png)
![Screenshot_3](https://user-images.githubusercontent.com/114281218/193944588-7c7038eb-97ca-4618-950a-ddd427ef833f.png)

**Reply by ard-av** [reply]
![Screenshot_4](https://user-images.githubusercontent.com/114281218/193944676-12434e56-ff9a-4087-9b69-795ea0a213e6.png)

**Reply by ard-av** [reply]
![Screenshot_5](https://user-images.githubusercontent.com/114281218/193944913-7fdb6974-1763-4e67-9efd-ce3062c3ad97.png)

**Reply by ard-av** [reply]
https://github.com/openai/whisper/discussions/130
C/C++ implementation of the model #130

**Reply by ard-av** [reply]
> Спасибо за видео - хорошая работа :)
> 
> Whisper, кажется, работает нормально (мне нужна была средняя модель для работы с датским, который довольно медленный, но работает нормально). Впечатляет со всеми языками!
> 
> Я пытался добавить поддержку Whisper в последнюю бета-версию SE: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip Должен работать со всеми файлами + выбранными строками через контекстное меню.
> 
> Время кажется немного неправильным по сравнению с Vosk/Kaldi. Кто-нибудь знает, можно ли получить по одному слову за раз?

https://github.com/openai/whisper/discussions/280
https://github.com/jianfch/stable-ts ???

**Reply by F-Emerick** [reply]
**Thank you!
I did it on windows and it works!
It had transcribed almost perfectly in portuguese.
Important:
I had to download git (https://scoop.sh/) and use powershell.
I had to use Python 3.10.8**


> I guess it's not for everybody - but try this:
> 
> 1. Download Python from: https://www.python.org/downloads/
> 2. Install pip: See https://phoenixnap.com/kb/install-pip-windows
> 3. Install whisper via cmd line, run: pip install git+https://github.com/openai/whisper.git
> 
> Models can be downloaded via the SE UI or will be auto downloaded when you run whisper.... I think.

**Comment by niksedk** [comment]
Try: whisper --model tiny.en --language English **--fp16 False** a.wav

**Reply by ard-av** [reply]
Thanks, it worked.

**Reply by ard-av** [reply]
I have already made several hours of Russian text on the Whisper model through Python. It's much better quality than the Vosk model.

**Reply by MbuguaDavid** [reply]
> Try: whisper --model tiny.en --language English **--fp16 False** a.wav

How can we achieve the --fp16 False in Subtitle Edit? I've got everything installed and ready to test it out but the "no text found" error is driving me crazy.

**Reply by ard-av** [reply]
I'm doing this not in SE yet, but in the Python terminal, as in the video from the first post. In SE I had the same error as you

**Reply by niksedk** [reply]
@MbuguaDavid: You can press F2 in the SE Whisper window to toogle log-view... `--fp16 False` should already be present (from a hidden `WhisperExtraSettings` in `Settings.xml`).
Latest beta might work better: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip

**Reply by MbuguaDavid** [reply]
> @MbuguaDavid: You can press F2 in the SE Whisper window to toogle log-view... `--fp16 False` should already be present (from a hidden `WhisperExtraSettings` in `Settings.xml`). Latest beta might work better: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip

It still doesn't work. I wish it did. The "No Text Found" Error still persists. :( 

https://user-images.githubusercontent.com/92556454/195743826-a96fb1cc-d8ab-4e4b-86b4-3045ef74df72.mp4

**Reply by shin3c** [reply]
@MbuguaDavid Try put ffmpeg.exe to "C:\Windows\system32" folder, or any folder that defined in "Environment Variables>PATH" It seems the problem was caused by Whisper couldn't find ffmpeg in your environment.

**Reply by coastal45** [reply]
That made no difference. But I added additional information on this problem in https://github.com/SubtitleEdit/subtitleedit/discussions/6328#discussioncomment-3963404

**Comment by coastal45** [comment]
@niksedk  Is it possible to find out the actual command that SE sends to whisper, and what result causes "No text found!" dialog?

**Reply by niksedk** [reply]
Pressing F2 might give the command line... or try to check the file "error_log.txt"
Also, latest beta of SE automatically uses ffmpeg if SE knows where it is (one less problem)

**Comment by coastal45** [comment]
@niksedk I did another test.
When running the sample file in SE (release 3.6.8), the F2 screen only shows the command sent to whisper, then nothing else until the window disappears and then "No text found!" dialog. 

**--EDIT--**

Then I entered the same whisper command at Windows Command Prompt. The result is attached. They don't display correctly in Command Prompt being non-ASCII, but copying to a text file displays Japanese characters.
Also, vtt/srt files were created, so it seems the basic whisper operation is ok.

Related files are attached that you can check with. Again, this is only happening in Windows, Linux works correctly.
On Windows 10, Python 3.10.8.

[files.zip](https://github.com/SubtitleEdit/subtitleedit/files/9942841/files.zip)

[files_srt.zip](https://github.com/SubtitleEdit/subtitleedit/files/9944409/files_srt.zip)
